### PR TITLE
Add a metric for monitoring how many evictions involve unexpired items

### PR DESCRIPTION
**Pull request by denkyl08**
There are certain cases where cache keys could be evicted while still being "unexpired" which would result in inaccurate rate limiting (see lru cache unit test in this MR). While this project prioritizes performance over accuracy, it would be good to know how often we're losing rate limit state so that we can more appropriately size the cache. 

In this MR I've added a new counter metric "gubernator_unexpired_evictions_count" to count the occurrence of this case.

### CODE REVIEW

File: lrucache_test.go [hunk 1]
```diff
@@ -28,8 +29,91 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+
+	dto "github.com/prometheus/client_model/go"
 )
 
+func TestLRUCacheUnexpiredEvictionMetrics(t *testing.T) {
```
Comment (Baliedge) [inline lrucache_test.go:36]: This test seems to have two test cases in one that should be separated:

1. Adding an item to full cache after some items have expired = no evictions.
2. Adding an item to full cache with no item expirations = 1 eviction.

**Review by Baliedge — CHANGES_REQUESTED** [review]
@denkyl08, once again thank you for your contribution.  I like the idea.  Please review my comments and let me know what you think.

File: lrucache.go [hunk 2]
```diff
@@ -53,6 +53,10 @@ var accessMetric = prometheus.NewCounterVec(prometheus.CounterOpts{
 	Name: "gubernator_cache_access_count",
 	Help: "Cache access counts.  Label \"type\" = hit|miss.",
 }, []string{"type"})
+var unexpiredEvictionsMetric = prometheus.NewCounter(prometheus.CounterOpts{
+	Name: "gubernator_unexpired_evictions_count",
+	Help: "Count the number of cache items which were evicted while unexpired",
```
[the remark this answers is not in the document]
Comment (denkyl08) [reply lrucache.go]: Fixed

File: lrucache_test.go (hunk 1 above)
Comment (denkyl08) [reply lrucache_test.go]: Updated to move this into two different t.Run blocks in TestLRUCache()

**Comment by Baliedge** [comment]
@denkyl08 Would you also do the honors of updating `CHANGELOG` as appropriate and call it rc.14?

**Comment by denkyl08** [comment]
@Baliedge Sure! I updated version file as well.

**Comment by Baliedge** [comment]
I had to do another RC release, so if you wouldn't mind updating your branch from `master` and set version to `rc.15`.  Also, there's some changes requested.  Once done, I'll approve and merge.

Thanks!

**Comment by denkyl08** [comment]
@Baliedge I've updated to rc15, and made the changes you requested in these comments: https://github.com/mailgun/gubernator/pull/129#pullrequestreview-879044981 Let me know if there's something else I need to do to mark these requested changes as completed. I'll admit that I'm still not accustomed to the github PR review UI.

**Comment by Baliedge** [comment]
@denkyl08 The last two code review comments for `lrucache.go` could use attention.  Also, if you can please avoid doing force push.  It messes up the code review and code browsing features.  I plan to do a squash and merge anyway, so it doesn't bother anybody if you have a million commits.

**Comment by denkyl08** [comment]
@Baliedge Understood. I will refrain from force pushing in the future. Can you clarify your comment on the two code review comments for `lrucache.go` I saw the one regarding punctuation of the metric description have addressed that. Is there another comment on `lrucache.go`?

**Review by Baliedge — APPROVED** [review]
My open comments can be addressed in a separate PR.  I approve the code as-is for now.

**Comment by thrawn01** [comment]
@denkyl08 This is a fantastic addition, I'm going to merge this and get it into the next release.

**Pull request merged**